Bug in optimizer with test_model and 3+ layers

Compiling a model with 3+ layers gives

```
@property
   def manager(self):
       “”"Return the GraphManager for this Graph.“”"
       if self._manager is None:
>           raise Exception(f’Graph {self} has no manager.’)
E           Exception: Graph 823:_graph0 has no manager.
```

while applying inline_trivial.

This used to work with a previous commit (but I didn't bisect to find it yet).

--- a/myia/opt/opt.py
+++ b/myia/opt/opt.py
@@ -230,7 +230,7 @@ class LocalPassOptimizer:
 
         while len(todo) > 0:
             n = todo.pop()
-            if n in seen:
+            if n in seen or n not in mng.all_nodes:
                 continue
             seen.add(n)
 

